### PR TITLE
Fix scheduler LSF driver logging error when killing finished jobs

### LINKED ISSUE #7658: LSF driver tries to kill failed realizations
Observed in the log

```
2024-04-12 12:52:17,533 - ert.scheduler.lsf_driver - LegacyEnsemble - DEBUG - Realization 9 (LSF-id: 7810) failed
2024-04-12 12:52:17,569 - ert.scheduler.lsf_driver - LegacyEnsemble - ERROR - LSF kill failed due to missing jobid for realization 9
```

This is analogous to #7560 

Make sure to unify the behaviour over all three drivers.

### PR DESCRIPTION
**Pull request by jonathan-eq**
**Issue**
Resolves #7658 

**Approach**
The commit in this PR fixes the bug where the scheduler LSF driver would log `LSF kill failed due to missing jobid for realization` when killing jobs that had already finished with a failure.

(Screenshot of new behavior in GUI if applicable)

- [ ] PR title captures the intent of the changes, and is fitting for release notes.
- [ ] Added appropriate release note label
- [ ] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).
- [ ] Make sure tests pass locally (after every commit!)

## When applicable
- [ ] **When there are user facing changes**: Updated documentation
- [ ] **New behavior or changes to existing untested code**: Ensured that unit tests are added (See [Ground Rules](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md#ground-rules)).
- [ ] **Large PR**: Prepare changes in small commits for more convenient review
- [ ] **Bug fix**: Add regression test for the bug
- [ ] **Bug fix**: Create Backport PR to latest release

### CODE REVIEW

File: src/ert/scheduler/lsf_driver.py [hunk 1]
```diff
@@ -265,6 +267,8 @@ async def submit(
         self._iens2jobid[iens] = job_id
 
     async def kill(self, iens: int) -> None:
+        if iens in self._finished_iens:
+            return
```
Comment (berland) [inline src/ert/scheduler/lsf_driver.py:271]: I think we are just covering an underlying problem with this, and should rather ensure we don't call `kill()` when we should not.

Comment (jonathan-eq) [reply src/ert/scheduler/lsf_driver.py:271]: Maybe we can move the `if iens in self._finished:...` check to before the place `driver.kill(...)` is called.

Comment (jonathan-eq) [reply src/ert/scheduler/lsf_driver.py:271]: Should the `finished` state be moved to the individual jobs (job.py) instead? That way we can check directly before calling `driver.kill(...)`

Comment (berland) [reply src/ert/scheduler/lsf_driver.py:271]: https://github.com/equinor/ert/pull/7660 is one possible solution that will fix this.

Comment (jonathan-eq) [reply src/ert/scheduler/lsf_driver.py:271]: @berland Great! Can I reassign you to this issue?

**Pull request closed without merging**